### PR TITLE
Launch SR health check on PBD.plug for clustered SRs

### PR DESCRIPTION
**Pull request by robhoes**

### CODE REVIEW

**Comment by robhoes** [comment]
The following PRs depend on each other:
* xapi-project/xen-api#2365
* xapi-project/xcp-idl#91
* xapi-project/xapi-storage#50
* xapi-project/xapi-storage-script#40

File: ocaml/xapi/xapi_pbd.ml [hunk 1]
```diff
@@ -115,6 +114,27 @@ let check_sharing_constraint ~__context ~self =
...
+	if Helpers.i_am_srmaster ~__context ~sr:self then
+		let dbg = Ref.string_of (Context.get_task_id __context) in
+		let info = C.SR.stat dbg (Db.SR.get_uuid ~__context ~self) in
+			if info.Storage_interface.clustered && info.Storage_interface.health = Storage_interface.Recovering then begin
+			Helpers.call_api_functions ~__context (fun rpc session_id ->
+				let task = Client.Client.Task.create ~rpc ~session_id ~label:"SR recovering"  ~description:"" in
+				let _ = Thread.create (fun () ->
+					let rec loop () =
+						Thread.delay 30.;
+						if info.Storage_interface.clustered && info.Storage_interface.health = Storage_interface.Recovering then
```
Comment (djs55) [inline ocaml/xapi/xapi_pbd.ml]: info is an immutable value at this point, so I think the expression will always evaluate to the same thing.

**Comment by djs55** [comment]
I think we might also need a way to shut the thread down on `PBD.unplug`.

**Comment by djs55** [comment]
I'd like to merge this as-is, since I'd like to build upon your SMAPIv* change. I think we can fix the issues with this separately.

File: ocaml/xapi/xapi_pbd.ml (hunk 1 above)
Comment (robhoes) [inline ocaml/xapi/xapi_pbd.ml]: Arggg... how stupid. I'll send an update shortly.

**Pull request merged**